blaze-client/src/main/scala/org/http4s/client/blaze/BlazeClientBuilder.scala

```diff
@@ -276,14 +276,33 @@ sealed abstract class BlazeClientBuilder[F[_]] private (
 
 object BlazeClientBuilder {
 
+  sealed trait SSLContextOption
+  object SSLContextOption {
+    case object NoSSL extends SSLContextOption
+    case object TryDefaultSSLContextOrNone extends SSLContextOption
```

Comment:
@kevinmeredith I believe `Try` already has the sense of `OrNone` (at least that it what it implies in the F# ecosystem)

Reply:
That's a good point. However, based on naming alone, it's not clear to me that `TryDefaultSSLContext` would default to `None` if the `Try` failed.

What are your thoughts, Keith and others?

Reply:
> at least that it what it implies in the F# ecosystem

At least from my experience in Scala, it's not clear to me that `Try` has the same implication in this language. However, I learn new things everyday, so Ross, Chris, et al, please let me know if the `OrNone` is unnecessary based on Keith's comment.

Reply:
The other reason I was thinking this was that you named this after an existing variable, which did not have `OrNone`. 🤷‍♂️

Reply:
> The other reason I was thinking this was that you named this after an existing variable, which did not have OrNone

Yeah, that's a good point, Keith. I'll re-name it to what you proposed. Thanks for pressing me.

Reply:
https://github.com/http4s/http4s/pull/3422/commits/7067e22da74a40fded2bbe54e7bdaae929b6b8fe